framework/src/multiapps/TransientMultiApp.C

```diff
@@ -418,21 +437,26 @@ TransientMultiApp::solveStep(Real dt, Real target_time, bool auto_advance)
                            << std::endl;
                 ex->incrementStepOrReject();
 
+                // Avoid numerical precision errors on target time
+                if (catch_up_time + catch_up_dt > dt)
+                  catch_up_dt = dt - catch_up_time;
```

Comment:
This looks strange to me - `catch_up_time` is always equal to zero here, right? If so, why include it? This looks like it would just be replaceable with `catch_up_dt = min(catch_up_dt, dt);`, for example.

Reply:
no it's not it builds up through the catch_up process. It could be for example:

catch_up_time = 0
catch_up_time = dt/4 (2 failures in a row)
catch_up_time = dt/2
and so on until we meet dt.

Reply:
I ve done this because the previous time check criterion relied on the two time scales being synchronized